game-core/src/main/java/games/strategy/engine/data/MutableProperty.java

```diff
@@ -92,11 +94,13 @@ public void setValue(final Object value) throws InvalidValueException {
   }
 
   private T cast(final Object value) throws InvalidValueException {
+    @SuppressWarnings("unchecked")
+    final Class<T> type = (Class<T>) typeToken.getRawType();
```

Comment:
I couldn't find a way to do the cast without an unchecked warning.  I can understand why `Class<? extends T>` may be incompatible with `Class<T>`, but, in this context, it seems safe since this class controls which `TypeToken<T>` was used at the time of creation.

This is in contrast to the unchecked warning we would get if we didn't use `Class#cast()` and did a simple `return (T) value` below.  In this case, this class has no control over `value`, a `ClassCastException` is completely reasonable, and thus the unchecked warning is not as easily justifiable to suppress.  As noted elsewhere, the problem with using a cast of this form is that the compiler will actually do the cast in the caller, and the particular `ClassCastException` handler in this method will be bypassed.  We can still handle it correctly by leaving this helper method intact but moving the `ClassCastException` handler to the caller in `setValue()`.